pyne/endl.py

```diff
@@ -0,0 +1,264 @@
+"""Module for parsing and manipulating data from ENDL evaluations.
+
+For the moment, classes and functions in this module only implement the
+specifications relevant to the EEDL (Evaluated Electron Data Library) and the
+EPDL (Evaluated Photon Data Library). The formats are described by the
+following documents:
+
+"ENDL type formats for the Livermore Evaluated Photon Data Library, EPDL"
+https://www.oecd-nea.org/dbdata/data/manual-endf/nds_eval_epdl.pdf
+
+"ENDL type formats for the Livermore Evaluated Electron Data Library, EEDL"
+https://www.oecd-nea.org/dbdata/data/manual-endf/nds_eval_eedl.pdf
+
+For more information, contact Davide Mancusi <davide.mancusi@cea.fr>.
+"""
+from __future__ import print_function, division, unicode_literals
+
+import re
+from warnings import warn
+from collections import namedtuple, defaultdict
+
+import numpy as np
+
+from pyne.utils import QAWarning
+from pyne import rxdata
+import pyne.utils as utils
+from pyne import nucname
+
+warn(__name__ + ' is not yet QA compliant.', QAWarning)
+
+END_OF_TABLE_RE = re.compile(' {71}1')
+
+
+class Library(rxdata.RxLib):
+    """A class for a file which contains multiple ENDL tables."""
+
+    DataTuple = namedtuple('DataTuple', ['yo', 'limits', 'x1', 'data'])
+
+    @staticmethod
+    def pin_rdesc_rprop_dict_entry():
+        """Static method to generate entries for the pin_rdesc_rprop dict."""
+
+        return {
+                'data_tuples': list()
+                }
+
+    @staticmethod
+    def structure_dict_entry():
+        """Static method to generate entries for the structure dict."""
+
+        return {
+                'pin': set(),
+                'rdesc': set(),
+                'rprop': set(),
+                'pin_rdesc_rprop': defaultdict(
+                    Library.pin_rdesc_rprop_dict_entry
+                    )
+                }
+
+    def __init__(self, fh):
+        self.structure = defaultdict(Library.structure_dict_entry)
+        self.intdict = {
+                0: self._linlin,
+                2: self._linlin,
+                3: self._loglin,
+                4: self._linlog,
+                5: self._loglog,
+                }
+        self.fh = fh
+        # read headers for all tables
+        self._read_headers()
+
+#    def load(self):
+#        """load()
+#        Read the ENDL file into a NumPy array.
+#
+#        Returns
+#        -------
+#        data : np.array, 1d, float64
+#            Returns a 1d float64 NumPy array.
+#        """
+#        opened_here = False
+#        if isinstance(self.fh, str):
+#            fh = open(self.fh, 'r')
+#            opened_here = True
+#        else:
+#            fh = self.fh
+#        fh.readline()
+#        data = fromendf_tok(fh.read())
+#        fh.seek(0)
+#        if opened_here:
+#            fh.close()
+#        return data
+
+    def _read_headers(self):
+        opened_here = False
+        if isinstance(self.fh, str):
```

Comment:
The str here probably needs to be python 3 compatible.  Look in other modules for code that looks like 

```python
if sys.version_info[0] > 2:
    basestring = str
```

and then test against basestring here.